engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/compile/ProcessCompiler.scala

```diff
@@ -331,12 +349,23 @@ protected trait ProcessCompilerBase {
                                       outputVariableNameOpt: Option[String],
                                       ctx: ValidationContext)
                                      (implicit nodeId: NodeId,
-                                      metaData: MetaData): ValidatedNel[ProcessCompilationError, T] = {
-    objectParametersExpressionCompiler.compileObjectParameters(nodeDefinition.parameters,
+                                      metaData: MetaData): (Map[String, ExpressionTypingInfo], ValidatedNel[ProcessCompilationError, T]) = {
+    val compiledObjectWithTypingInfo = objectParametersExpressionCompiler.compileObjectParameters(nodeDefinition.parameters,
       parameters,
       branchParameters, ctx.clearVariables, ctx).andThen { compiledParameters =>
       try {
-        Valid(factory.create[T](nodeDefinition, compiledParameters, outputVariableNameOpt))
+        val obj = factory.create[T](nodeDefinition, compiledParameters, outputVariableNameOpt)
```

Comment:
can we extract it somewhere?

Reply:
You mean `factory.create` + surrounding try/catch? Good idea, will do that.